src/Workspaces/CSharp/Portable/LanguageServices/CSharpTypeInferenceService.TypeInferrer.cs

```diff
@@ -1950,6 +1951,121 @@ private IEnumerable<TypeInferenceInfo> InferTypeInVariableDeclarator(VariableDec
                 return types;
             }
 
+            private IEnumerable<TypeInferenceInfo> InferTypeInVariableComponentAssignment(
+                VariableComponentAssignmentSyntax variableComponentAssigment, ExpressionSyntax expression)
+            {
+                if (expression == variableComponentAssigment.Value)
+                {
+                    var variableComponent = variableComponentAssigment.VariableComponent;
+                    if (variableComponent.IsKind(SyntaxKind.TypedVariableComponent))
+                    {
+                        var typedVariable = (TypedVariableComponentSyntax)variableComponent;
```

Comment:
Comment for the code this case represents? This class is pretty good about having examples.